src/t_stream.c

```diff
@@ -2880,68 +2914,85 @@ void xpendingCommand(client *c) {
     long long count = 0;
     long long minidle = 0;
     int startex = 0, endex = 0;
+    int j = 3;
+    int mkgroup = 0;
 
-    /* Start and stop, and the consumer, can be omitted. Also the IDLE modifier. */
-    if (c->argc != 3 && (c->argc < 6 || c->argc > 9)) {
+    if (c->argc > 10) {
         addReplyErrorObject(c,shared.syntaxerr);
         return;
     }
 
-    /* Parse start/end/count arguments ASAP if needed, in order to report
-     * syntax errors before any other error. */
-    if (c->argc >= 6) {
-        int startidx = 3; /* Without IDLE */
-
-        if (!strcasecmp(c->argv[3]->ptr, "IDLE")) {
-            if (getLongLongFromObjectOrReply(c, c->argv[4], &minidle, NULL) == C_ERR)
+    if (c->argc >= 4) {
+        if (!strcasecmp(c->argv[j]->ptr, "MKGROUP")) {
+            mkgroup = 1;
+            j++;
+        }
+        if (!strcasecmp(c->argv[j]->ptr, "IDLE")) {
+            if (j+1 == c->argc) {
+                addReplyErrorObject(c,shared.syntaxerr);
+                return;
+            }
+            if (getLongLongFromObjectOrReply(c, c->argv[j+1], &minidle, NULL) == C_ERR)
                 return;
             if (c->argc < 8) {
```

Comment:
isn't this outdated? let's match it to `j`

Reply:
adressed this comment when i updated following comment https://github.com/redis/redis/pull/10747#discussion_r1063563264